### PR TITLE
Time mechanism for minimax

### PR DESCRIPTION
**Pull request by jonathanjosef91**

### CODE REVIEW

File: wet2/main.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import argparse
+import time
```
Comment (jonathanjosef91) [inline wet2/main.py:2]: delete

File: wet2/players/MinimaxPlayer.py [hunk 2]
```diff
@@ -38,9 +42,34 @@ def make_move(self, time_limit):
         output:
             - direction: tuple, specifing the Player's movement
         """
-        step = self.minimax.search(self.currentState, 2, True)[1]
-        self.currentState.updatePlayerStep(step)
-        return step.getStep()
+        startTime = time.time()
+        deadline = startTime + time_limit/2
```
Comment (jonathanjosef91) [inline wet2/players/MinimaxPlayer.py:46]: Margin too wide

File: wet2/players/MinimaxPlayer.py [hunk 3]
```diff
@@ -38,9 +42,34 @@ def make_move(self, time_limit):
...
+            except TimeoutException:
+                print('Timeout')
+                break
+
+            depth += 1
+            lastRun = time.time() - endTime
+            endTime = time.time()
+            print(depth, lastRun)
+
+        print('depth reached: ', depth)
```
Comment (jonathanjosef91) [inline wet2/players/MinimaxPlayer.py:65]: Add TODO to delete before sub

File: wet2/utils.py [hunk 4]
```diff
@@ -106,6 +106,11 @@ def __eq__(self, other):
                 and self.dead == other.dead
                 and self.isPlayer == other.isPlayer)
 
+    def __gt__(self, other):
+        return (self.pos < other.pos
+                or (self.pos == other.pos and self.soldier < other.soldier)
+                or (self.pos == other.pos and self.soldier < other.soldier and self.dead < other.dead))
```
Comment (jonathanjosef91) [inline wet2/utils.py:112]: self.soldier == other.soldier

**Review by tayaharshuk1 — APPROVED** [review]

**Pull request merged**